### DISCUSSION TITLE
Asset editor - LHT vs. RHT

### DISCUSSION THREAD

**Discussion by originalfoo** [CitiesSkylinesMods/TMPE#1264] [Q&A]
@kianzarrin If an intersection is customised with TM:PE in the asset/map editor, what happens if subscribers to that map/asset set their traffic to drive on the other side of road?

Example:

* Map author creates RHT map and customises intersections on that map using TMPE.
* User subscribes the map, but enables `Traffic drives on left` when starting their new city

What happens to the TMPE customisations now the roads are reversed?

**Comment by kianzarrin** [comment]
I think they all break. Even if you just invert a road using the toggle tool it might break lane connections (I mean they are visually there but they don't seem to work).

**Reply by kianzarrin** [reply]
something to think about with regards to persistency.

**Reply by kianzarrin** [reply]
as for the LHT/RHT switch I don't know if it is even possible to maintain some TMPE rules in a sensible way.

**Reply by kianzarrin** [reply]
see https://github.com/CitiesSkylinesMods/TMPE/issues/1270

**Reply by originalfoo** [reply]
If we're going to mention asset editor support in 11.6 release, we need a solution for that problem IMO - or at least let asset creators know that their asset will break if end-user city drives on other side of road

**Reply by originalfoo** [reply]
Also which features exactly are supported in the asset editor?

- [x] Parking Restrictions
- [x] Speed Limits (and default speeds?)
- [x] Vehicle Restrictions
- [x] Junction Restrictions
- [x] Lane Arrows
- [x] Lane Connectors
- [x] Priority Signs
- [ ] Timed Traffic Lights

I plan on creating a wiki page about asset editor support, and in there we could mention the LHT/RHT issue.

**Reply by kianzarrin** [reply]
all except TTL

**Reply by originalfoo** [reply]
I assume there's some sort of version number in the data just in case stuff changes at later date?

BTW, does 'export' feature of Move It also clone those settings (and thus if user shares their export what happens when another users imports it to a city that drives on the opposite side)??

**Reply by originalfoo** [reply]
I guess what I'm trying to say is that we should probably include value of `Singleton<SimulationManager>.instance.m_metaData.m_invertTraffic` with the data so that should the city it ends up in be driving on other side it will be easier to just bulk ignore the majority of settings in the asset/map/import.

**Reply by kianzarrin** [reply]
there is version.
Move it clones all those settings

> what happens when another users imports it to a city that drives on the opposite side
 
we have to test that scenario.

maybe moveit can provide this info.

**Reply by kianzarrin** [reply]
or maybe we can store in the lane record to so that if someone changes the direction of the highway, we would know what to do.

**Reply by kianzarrin** [reply]
now that I think about it bi-directional lane connections does have some advantages!
